src/Microsoft.Extensions.Configuration.AzureAppConfiguration/Extensions/TimeSpanExtensions.cs

```diff
@@ -32,5 +32,27 @@ public static TimeSpan CalculateBackoffTime(this TimeSpan interval, int attempts
 
             return TimeSpan.FromTicks(Math.Min(maxBackoff.Ticks, calculatedBackoff.Ticks));
         }
+
+        /// <summary>
+        /// This method calculates a random exponential time to retry the primary store after a failure
+        /// which lies between <paramref name="interval"/> and <see cref="RefreshConstants.DefaultMaxRetryAfter"/>.
+        /// </summary>
+        /// <param name="interval">The minimum interval to retry after.</param>
+        /// <param name="attempts">The number of attempts made to the primary config store.</param>
+        /// <returns>The calculated exponential time to retry the primary config store after a failure.</returns>
+        /// <exception cref="ArgumentOutOfRangeException">
+        /// An exception is thrown when <paramref name="attempts"/> is less than 1.
+        /// </exception>
+        public static TimeSpan CalculateRetryAfterTime(this TimeSpan interval, int attempts)
+        {
+            if (attempts < 1)
+            {
+                throw new ArgumentOutOfRangeException(nameof(attempts), attempts, "The number of attempts should not be less than 1.");
+            }
+
+            TimeSpan calculatedRetryAfter = TimeSpan.FromTicks(interval.Ticks * new Random().Next(1, (int)Math.Min(Math.Pow(2, attempts - 1), int.MaxValue)));
```

Comment:
I think it would be a better idea to use double to progress the exponential backoff. Otherwise we work with discrete increments of the input interval rather than continuous random backoff.

Reply:
I'm confused. Are you suggesting we use `double` instead of `int` when requesting a random value?
`Random.nextDouble()` would return a next random double value, but it doesn't have an overload which will accept a range and provide a random double in that range.
Can you elaborate little more on this?

Reply:
With the latest design, we are just doubling the interval with each failed attempt, so don't need a random anymore.